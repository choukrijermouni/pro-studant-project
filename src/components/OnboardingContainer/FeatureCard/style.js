import {
  Spacing,
  neutral
} from '@pro_boa/ui'

export default {
  userIcon: {
    display: 'flex',
    margin: () => Spacing(0, 3, 0, 0),
    alignItems: 'center'
  },
  paper: {
    width: ({ width }) => width || 368,
    display: 'flex',
    alignItems: 'center',
    height: 119
  },
  icon: {
    height: '61%',
    background: neutral[0],
    display: 'flex',
    width: 115,
    justifyContent: 'center',
    fontSize: 35,
    padding: 22,
    zIndex: 1,
    borderRadius: '4px 0 0 4px',
    backgroundColor: neutral[0],
    boxShadow: '0 0 36px 0 rgb(194 212 230 / 23%)'
  },
  text: {
    width: '71%',
    height: '100%',
    background: ({ cardColor }) => cardColor,
    display: 'flex',
    alignItems: 'center',
    transition: 'background .5s linear'
  },
  textActive: {
    background: `${neutral[0]} !important`,
    boxShadow: ({ cardColor }) => `inset 0 -2px 0 0 ${cardColor}, 0 0 36px 0 rgba(194,212,230,0.23)`
  }
}
