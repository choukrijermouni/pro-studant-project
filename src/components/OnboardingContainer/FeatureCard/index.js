import {
  Paper,
  TypographyElement,
  neutral
} from '@pro_boa/ui'
import classNames from 'classnames'
import { useState } from 'react'
import { createUseStyles } from 'react-jss'
import style from './style'
const useStyles = createUseStyles(style)

export default ({ cardIcon, context, cardColor, width }) => {
  const [active, setActive] = useState(true)
  const { userIcon, paper, icon, text, textActive } = useStyles({ cardColor, width })
  return (
    <div onClick={() => setActive(!active)}>
      <Paper className={paper}>
        <div className={icon}>
          <img src={cardIcon} alt='logo' className={userIcon} />
        </div>
        <div className={classNames(active && textActive, text)}>
          {active &&
            <TypographyElement
              component='h4'
              variant='heading4'
              align='left'
              spacing='50px 24px'
              color={active ? cardColor : neutral[0]}
            >
              {context?.title}
            </TypographyElement>}
          {!active &&
            <div>
              <TypographyElement
                component='h4'
                variant='heading4'
                align='left'
                spacing='21px 24px 0'
                color={active ? cardColor : neutral[0]}
              >
                {context.title}
              </TypographyElement>
              <TypographyElement
                component='h4'
                variant='heading4'
                align='left'
                fontWeight={200}
                spacing='8px 24px 21px'
                color={neutral[0]}
              >
                {context.body}
              </TypographyElement>
            </div>}
        </div>
      </Paper>
    </div>
  )
}
