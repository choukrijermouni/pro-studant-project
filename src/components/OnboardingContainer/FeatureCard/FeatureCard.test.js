import { shallow, configure } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import renderer from 'react-test-renderer'
import FeatureCard from '.'
configure({ adapter: new Adapter() })

describe('>>> FeatureCard --- Shallow Render ManagerCard component', () => {
  let wrapper
  let tree

  beforeEach(() => {
    wrapper = shallow(<FeatureCard />)
    tree = renderer.create(
      <FeatureCard />
    ).toJSON()
  })

  it('+++ render correctly and match snapshot', () => {
    expect(tree).toMatchSnapshot()
  })

  it('+++ render the component', () => {
    expect(wrapper).toBeTruthy()
  })
})
