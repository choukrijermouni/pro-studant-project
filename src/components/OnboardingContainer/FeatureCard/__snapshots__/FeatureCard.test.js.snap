// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> FeatureCard --- Shallow Render ManagerCard component +++ render correctly and match snapshot 1`] = `
<div
  onClick={[Function]}
>
  <div
    className="paper-0-2-2 paper-d5-0-2-11 paper-0-2-14"
  >
    <div
      className="icon-0-2-3"
    >
      <img
        alt="logo"
        className="userIcon-0-2-1 userIcon-d4-0-2-10"
      />
    </div>
    <div
      className="textActive-0-2-5 textActive-d7-0-2-13 text-0-2-4 text-d6-0-2-12"
    >
      <p
        className="text-0-2-15 text-d0-0-2-16"
      />
    </div>
  </div>
</div>
`;
